File: packages/bot_tele/session.py
from telegram import User

users = {}


async def init_session(user: User) -> None:
    user_id = str(user.id)
    if not user_id in users:
        users[user_id] = {
            "user_id": user_id,
            "full_name": user.full_name,
        }
    return users[user_id]

async def get_session(user: User): 
    if not str(user.id) in users:
        await init_session(user)
    return users[str(user.id)]

async def update_session(user: User) -> None:
    user_id = str(user.id)
    if not user_id in users:
        await init_session(user)
    users[user_id] = {**users[user_id] }
    return users[user_id]

File: packages/bot_wha/session.ts
// Description: This file contains the session management for the bot.

import { TSession, TUserContext, TUser } from "./types"


const users: TSession = {}


export const init_session = (user_data: TUser): TUserContext => {
    const user_id: string = user_data["user_id"]
    users[user_id] = {
        ...user_data,
        "context": null
    }
    return users[user_id]
}

export const get_session = (user_data: TUser): TUserContext => {
    const user_id: string = user_data["user_id"]
    if (!users[user_id]) {
        return init_session(user_data)
    }
    return users[user_id] as TUserContext
}

export const update_session = (user_data: TUser, context: string): TUserContext => {
    const user_id: string = user_data["user_id"]
    users[user_id] = {
        ...users[user_id],
        ...user_data,
        context
    }
    return users[user_id]
}
